File: 06-DataBinding-VB/MainForm.vb
Imports System.Windows.Forms

''' <summary>
''' Launches the two data binding scenarios: binding the editor's BodyHtml property, and
''' binding its DocumentHtml property, to a field on a bound business object.
'''
''' The layout lives in the designer; open MainForm.vb in the Visual Studio designer to see it.
''' </summary>
Partial Public Class MainForm
    Inherits Form

    Public Sub New()
        InitializeComponent()

        ' No license key set, so the editor runs in trial mode. See the licensing docs linked in the README.
    End Sub

    Private Sub OnBodyHtmlButtonClick(sender As Object, e As EventArgs) Handles _bodyHtmlButton.Click
        Dim demoForm As New BodyHtmlBindingForm()
        demoForm.Show()
    End Sub

    Private Sub OnDocumentHtmlButtonClick(sender As Object, e As EventArgs) Handles _documentHtmlButton.Click
        Dim demoForm As New DocumentHtmlBindingForm()
        demoForm.Show()
    End Sub

End Class

File: 06-DataBinding-CS/MainForm.cs
namespace DataBinding;

/// <summary>
/// Launches the two data binding scenarios: binding the editor's BodyHtml property, and
/// binding its DocumentHtml property, to a field on a bound business object.
///
/// The layout lives in the designer; open MainForm.cs in the Visual Studio designer to see it.
/// </summary>
public partial class MainForm : Form
{
    public MainForm()
    {
        InitializeComponent();

        // No license key set, so the editor runs in trial mode. See the licensing docs linked in the README.
    }

    private void OnBodyHtmlButtonClick(object? sender, EventArgs e)
    {
        new BodyHtmlBindingForm().Show();
    }

    private void OnDocumentHtmlButtonClick(object? sender, EventArgs e)
    {
        new DocumentHtmlBindingForm().Show();
    }
}
